client/state/products-list/selectors/is-marketplace-product.js

```diff
@@ -0,0 +1,13 @@
+import 'calypso/state/products-list/init';
+
+const productSlugToSubscriptionSlug = ( productSlug ) => productSlug.replace( '-', '_' );
+
+export const isMarketplaceProduct = ( state, productSlug ) =>
+	Boolean(
+		Object.entries( state.productsList?.items ).find(
+			( [ slug, { product_type } ] ) =>
+				( slug.startsWith( productSlug ) ||
+					slug.startsWith( productSlugToSubscriptionSlug( productSlug ) ) ) &&
```

Comment:
It seems to me that this will return true when searching for `woocommerce_test_plugin` but the actual marketplace product is `woocommerce_test_plugin_advanced`?

Reply:
Maybe it also worths adding a test for this case.

Reply:
That's right, I haven't thought of this case before. I'll add some tests to cover this too.

Reply:
Added some tests to cover this scenario 👍